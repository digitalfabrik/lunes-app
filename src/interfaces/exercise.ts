export interface IVocabularyOverviewScreen {
  navigation: any
  route: any
}

export interface IVocabularyTrainerScreen {
  navigation: any
  route: any
}

export interface IAlternativeWordProps {
  alt_word: string
  article: string
}

export interface IDocumentProps {
  id: number
  word: string
  article: string
  document_image: Array<{ id: number; image: string }>
  audio: string
  alternatives: IAlternativeWordProps[]
}

export interface IVocabularyOverviewListItemProps {
  id: number
  word: string
  article: string
  image: string
  audio: string
}

export interface IConfirmationModalProps {
  visible: boolean
  setIsModalVisible: Function
  navigation: any
  extraParams: any
}

export interface IAnswerSectionProps {
  tryLater: () => void
  currentDocumentNumber: number
  setCurrentDocumentNumber: Function
  documents: IDocumentProps[]
  finishExercise: Function
  trainingSet: string
  disciplineTitle: string
}

export interface IFeedbackProps {
  secondAttempt: boolean
  result: string
  document?: IDocumentProps
  input: string
}

export interface IActionsProps {
  tryLater: () => void
  giveUp: () => void
  result: string
  checkEntry: () => void
  getNextWord: () => void
  input: string
  isFinished: boolean
  secondAttempt: boolean
}
