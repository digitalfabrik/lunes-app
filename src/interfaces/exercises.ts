export interface IExercisesScreenProps {
  route: any
  navigation: any
}
