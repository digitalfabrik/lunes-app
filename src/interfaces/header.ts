export interface IHeaderProps {
  top: number | undefined
}
