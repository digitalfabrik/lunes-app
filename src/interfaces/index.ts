import { ReactElement } from 'react'

export interface ILoadingProps {
  children: ReactElement
  isLoading: boolean
}

export interface IButtonProps {
  onPress: () => void
  disabled?: boolean
  children: ReactElement
  theme?: string
  testID?: string
}
export interface IPopoverProps {
  setIsPopoverVisible: (param: boolean) => void
  isVisible: boolean
  children: ReactElement
}
