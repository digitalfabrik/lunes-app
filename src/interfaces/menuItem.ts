import { ReactElement } from 'react'

export interface IMenuItemProps {
  selected: boolean
  onPress: () => void
  icon: string
  title: string
  children: ReactElement
}
