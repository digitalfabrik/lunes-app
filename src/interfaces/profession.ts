export interface IProfessionScreenProps {
  navigation: any
}

export interface IProfessionsProps {
  id: number
  title: string
  description: string
  icon?: string
  total_training_sets: number
}
