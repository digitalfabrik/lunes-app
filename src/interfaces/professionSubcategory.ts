export interface IProfessionSubcategoryScreenProps {
  route: any
  navigation: any
}

export interface IProfessionSubcategoryProps {
  id: number
  title: string
  description: string
  icon: string
  total_documents: number
}
