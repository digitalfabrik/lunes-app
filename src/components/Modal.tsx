import React from 'react'
import { View, Text, TouchableOpacity, Modal, StyleSheet } from 'react-native'
import { IConfirmationModalProps } from '../interfaces/exercise'
import { CloseIcon } from '../../assets/images'
import Button from './Button'
import { BUTTONS_THEME } from '../constants/data'
import AsyncStorage from '@react-native-async-storage/async-storage'
import { COLORS } from '../constants/colors'
import { widthPercentageToDP as wp } from 'react-native-responsive-screen'

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  overlay: {
    marginTop: 0,
    backgroundColor: COLORS.lunesOverlay
  },
  modal: {
    backgroundColor: COLORS.white,
    alignItems: 'center',
    justifyContent: 'center',
    width: wp('85%'),
    borderRadius: 4,
    position: 'relative',
    paddingVertical: 31
  },
  closeIcon: {
    position: 'absolute',
    top: 8,
    right: 8,
    width: 24,
    height: 24
  },
  message: {
    textAlign: 'center',
    fontSize: wp('5%'),
    color: COLORS.lunesGreyDark,
    fontFamily: 'SourceSansPro-SemiBold',
    width: wp('60%'),
    marginBottom: 31,
    paddingTop: 31
  },
  lightLabel: {
    color: COLORS.lunesWhite,
    fontSize: wp('4%'),
    fontWeight: '600',
    textAlign: 'center',
    textTransform: 'uppercase',
    fontFamily: 'SourceSansPro-SemiBold',
    letterSpacing: 0.4
  },
  darkLabel: {
    color: COLORS.lunesBlack,
    fontSize: wp('4%'),
    fontWeight: '600',
    textAlign: 'center',
    textTransform: 'uppercase',
    fontFamily: 'SourceSansPro-SemiBold',
    letterSpacing: 0.4
  }
})

const ConfirmationModal = ({ visible, setIsModalVisible, navigation, extraParams }: IConfirmationModalProps) => {
  const closeModal = () => setIsModalVisible(false)

  const goBack = () => {
    setIsModalVisible(false)
    AsyncStorage.removeItem('session')
    navigation.navigate('Exercises', { extraParams })
  }

  return (
    <Modal testID='modal' visible={visible} transparent animationType='fade' style={styles.container}>
      <View style={[styles.container, styles.overlay]}>
        <View style={styles.modal}>
          <TouchableOpacity style={styles.closeIcon} onPress={closeModal}>
            <CloseIcon />
          </TouchableOpacity>
          <Text style={styles.message}>Do you really want to end this session?</Text>
          <Button onPress={closeModal} theme={BUTTONS_THEME.dark}>
            <Text style={styles.lightLabel}>continue</Text>
          </Button>

          <Button onPress={goBack} theme={BUTTONS_THEME.light}>
            <Text style={styles.darkLabel}>end</Text>
          </Button>
        </View>
      </View>
    </Modal>
  )
}

export default ConfirmationModal
