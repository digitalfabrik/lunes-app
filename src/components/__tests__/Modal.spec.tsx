import { View } from 'react-native'
import React from 'react'
import Modal from '../Modal'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { IConfirmationModalProps } from '../../interfaces/exercise'

describe('Components', () => {
  describe('Modal ', () => {
    const defaultModalProps: IConfirmationModalProps = {
      navigation: '',
      setIsModalVisible: () => {},
      visible: false
    }

    it('renders correctly across screens', () => {
      const component = shallow(
        <View>
          <Modal {...defaultModalProps} />
        </View>
      )
      expect(toJson(component)).toMatchSnapshot()
    })

    it('should have visible property passed to it as default', () => {
      const modalProps: IConfirmationModalProps = {
        ...defaultModalProps
      }

      const component = shallow(
        <View>
          <Modal {...modalProps} />
        </View>
      )
      expect(component.children().props().visible).toBe(false)
    })

    it('should have visible property passed to it', () => {
      const modalProps: IConfirmationModalProps = {
        ...defaultModalProps,
        visible: true
      }

      const component = shallow(
        <View>
          <Modal {...modalProps} />
        </View>
      )
      expect(component.children().props().visible).toBe(true)
    })
  })
})
