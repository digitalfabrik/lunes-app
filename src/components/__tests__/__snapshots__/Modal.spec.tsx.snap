// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Modal  renders correctly across screens 1`] = `
<View>
  <ConfirmationModal
    navigation=""
    setIsModalVisible={[Function]}
    visible={false}
  />
</View>
`;
